src/core/generator/from_proto.rs

```diff
@@ -273,6 +273,12 @@ impl Context {
                 // required only applicable for proto2
                 cfg_field.required = matches!(label, Label::Required);
 
+                if cfg_field.list {
+                    // by default proto3 fields are optional and default values are provided.
+                    // in order to support proto2, we mark these list fields required.
+                    cfg_field.required = true;
```

Comment:
it's acceptable to mark the field as required for output types, but for input types it adds a stricter requirement than defined by protobuf which may be not desired.

we have todo like this https://github.com/tailcallhq/tailcall/blob/64e3c53747c7a6a47eebff74a5b85a6ea17f19e4/src/core/blueprint/operators/grpc.rs#L81-L84
and I generally think that json_schema validation should not enforce the nullability of the fields at all:
- if you define the field as optional in graphql everything is fine because fields in proto3 are optional
- if you define the field as required in graphql you just enforcing rules on the graphql side and everything still works on grpc side
- for proto2 field should be marked as required if it's required and therefore we need to validate that graphql also defines it as required